muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/forum/rest/WorkspaceForumRESTService.java

```diff
@@ -399,16 +400,18 @@ public Response updateThread(@PathParam ("WORKSPACEENTITYID") Long workspaceEnti
     }
     
     if (sessionController.hasPermission(MuikkuPermissions.OWNER, forumThread) || sessionController.hasWorkspacePermission(ForumResourcePermissionCollection.FORUM_EDIT_WORKSPACE_MESSAGES, workspaceEntity)) {
-      if (!forumThread.getSticky().equals(updThread.getSticky()) || !forumThread.getLocked().equals(updThread.getLocked())) {
+      if (!forumThread.getSticky().equals(updThread.getSticky()) || String.valueOf(forumThread.getLocked()) != updThread.getLock()) {
```

Comment:
I'm not sure how String.valueOf(enum) behaves (enums are tricky sometimes) so I would approach this by taking the payload lock value to a variable and then use it here and later for `updateForumThread`.

`LockForumThread lock = updThread.getLock() != null ? LockForumThread.valueOf(updThread.getLock()) : null;`
...
`if ( ... forumThread.getLocked() != lock)`

and use lock variable value in updateForumThread too